shunting-yard.h

```diff
@@ -188,18 +188,43 @@ struct rpnBuilder {
   // * * * * * Static parsing helpers: * * * * * //
 
   // Check if a character is the first character of a variable:
-  static inline bool isvarchar(const char c) {
-    return isalpha(c) || c == '_';
+  // Returns the byte-length of the character
+  // (rest is needed for UTF8 characters)
+  static inline unsigned char isvarchar(const char c, const char** rest) {
+    unsigned char utf8charsize = isUTF8char(c, rest);
+    if(utf8charsize == 0) return isalpha(c) || c == '_' ? true : false;
+    return utf8charsize;
+  }
+
+  // Checks if this is the start of a multi-character unicode character
+  // and if it is, will return the byte-size of the character.
+  // returns zero if it is not a multi-character unicode character
+  // throws a domain_error exception if the character is malformed
+  static inline unsigned char isUTF8char(const char c, const char** rest) {
+    unsigned char counter = 0;
+    if(c & 0x80) { // This is the start of a multi-char unicode character
+      counter++;
+      if(c & 0x40) counter++;
+      if(c & 0x20) counter++;
+      if(c & 0x10) counter++;
+      for(int i = 0; i < counter; i++) {
+        if(!((*rest)[i] & 0x80))
+          throw std::domain_error("Subsequent bytes of unicode character have to be of the form \\b10xxxxxx");
+      }
+    }
+    return counter;
   }
 
   static inline std::string parseVar(const char* expr, const char** rest = 0) {
     std::stringstream ss;
-    ss << *expr;
-    ++expr;
-    while (rpnBuilder::isvarchar(*expr) || isdigit(*expr)) {
-      ss << *expr;
-      ++expr;
-    }
+    unsigned char charsize;
+    do {
+        charsize = isvarchar(*expr, rest) + isdigit(*expr);
+        for (int i = 0; i < charsize; i++) {
+            ss << *expr++;
+            *rest = expr;
```

Comment:
null (0) is a valid value for this variable, since it is optional. Also since this is a output variables, we have no need to set it on every iteration, we can leave it to be updated on line 228 below (`if (rest) *rest = expr;`)

Reply:
Also @sillydan1 if you give me permission to push to your master branch, I can try to tackle this function, I have an idea of a simple implementation that might solve it.

Reply:
Woah! Thanks for all the feedback! I've been pretty busy myself, so I'm just now reading your comments. I'll give you permission to push asap :smile: